### PR TITLE
Remove bad restriction for default strings

### PR DESCRIPTION
**Pull request by saicheems**
There is no requirement that path template patterns have an even number
of segments.

### CODE REVIEW

**Comment by saicheems** [comment]
This adds support for cases where path templates terminate in a literal instead of a wildcard.

See:
https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/discoveries/genomics.v1.json#L2824

or 

https://github.com/googleapis/discovery-artifact-manager/blob/master/discoveries/dataproc.v1.json#L597

File: src/main/java/com/google/api/codegen/discovery/DefaultString.java [hunk 1]
```diff
@@ -94,14 +94,22 @@ static String forPattern(String pattern, String placeholderFormat) {
     }
 
     StringBuilder ret = new StringBuilder();
```
Comment (tcoffee-google) [inline src/main/java/com/google/api/codegen/discovery/DefaultString.java]: What does this variable name mean? Maybe change to something more obvious.

File: src/main/java/com/google/api/codegen/discovery/DefaultString.java [hunk 2]
```diff
@@ -94,14 +94,22 @@ static String forPattern(String pattern, String placeholderFormat) {
...
-    for (int i = 0; i < elems.size(); i += 2) {
-      String literal = elems.get(i).getLiteral();
-      String placeholder = Inflector.singularize(literal);
-      placeholder = LanguageUtil.lowerCamelToLowerUnderscore(placeholder).replace('_', '-');
-      ret.append('/')
-          .append(literal)
-          .append("/")
-          .append(String.format(placeholderFormat, placeholder));
+    for (int i = 0; i < elems.size(); i++) {
```
Comment (tcoffee-google) [inline src/main/java/com/google/api/codegen/discovery/DefaultString.java]: Given the revised specification, it seems more appropriate to keep the looping by two, and simply add an additional check at the end for an extra literal.

File: src/main/java/com/google/api/codegen/discovery/DefaultString.java (hunk 1 above)
Comment (saicheems) [inline src/main/java/com/google/api/codegen/discovery/DefaultString.java]: Changed to `defaultString`

File: src/main/java/com/google/api/codegen/discovery/DefaultString.java (hunk 2 above)
Comment (saicheems) [inline src/main/java/com/google/api/codegen/discovery/DefaultString.java]: Done

**Pull request merged**